File: 2.1.H/Python/system.py
class System():
    def __init__(self, MatchStragety):
        self.individuals = list()
        self.matchStragety = MatchStragety()

    def setIndividuals(self, individuals):
        self.individuals = individuals
    
    def getIndividuals(self):
        return self.individuals

    def match(self, individual):
        result = self.matchStragety.match(self.individuals, individual)[0]
        individual.setIndividual(result)
        return individual

    
    
           

File: 2.1.H/Java/StrategyPattern/src/System.java
import java.util.List;

public class System {
    private List<Individual> individuals;
    private MatchStragety matchStragety;
    
    public void setIndividuals(List<Individual> individuals) throws IllegalArgumentException {
        if (individuals == null || individuals.size() < 2) {
            throw new IllegalArgumentException("Individuals must be greater than 2");
        }
        this.individuals = individuals;
    }

    public List<Individual> getIndividuals() {
        return this.individuals;
    }

    public void match(Individual targetIndividual) throws Exception {
        if (targetIndividual == null) {
            throw new Exception("Target individual must not be null");
        }
        
        targetIndividual.setIndividual(matchStragety.match(individuals, targetIndividual).get(0));

    }
}
